File: Python/QualityAdjustedLifeYear.py
'''
Problem: https://open.kattis.com/problems/qaly
Author:  Bivash Pandey
'''
numOfPeriods = int(input())
qaly = 0.0
for i in range(numOfPeriods):
    qualityOfLife, years = input().split(" ")
    qaly += float(qualityOfLife) * float(years)
print(f'{qaly:.3f}')

File: Java/QualityAdjustedLifeYear.java
/*
* Problem: https://open.kattis.com/problems/qaly
* 
* @author  Bivash Pandey
*/

import java.util.Scanner;

public class QualityAdjustedLifeYear {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int numOfPeriods = input.nextInt();
        double qaly = 0.0;
        for(int i=1; i<=numOfPeriods; i++) {
            double qualityOfLife = input.nextDouble();
            double years = input.nextDouble();
            qaly += qualityOfLife * years;
        }
        System.out.println(String.format("%.3f", qaly));
    }
}
